### PR TITLE
Document the functions whitelist option in AbstractFunctionRestrictionsSniff

### PR DESCRIPTION
**Pull request by jrfnl**
This feature was added in #626, but not properly documented so far.

### CODE REVIEW

File: WordPress/AbstractFunctionRestrictionsSniff.php [hunk 1]
```diff
@@ -61,11 +61,16 @@
 	 * 	'lambda' => array(
 	 * 		'type'      => 'error' | 'warning',
 	 * 		'message'   => 'Use anonymous functions instead please!',
-	 * 		'functions' => array( 'eval', 'create_function' ),
+	 * 		'functions' => array( 'eval', 'create_function', 'mysql_*' ),
+	 *      // Only useful when using wildcards:
```
Comment (grappler) [inline WordPress/AbstractFunctionRestrictionsSniff.php]: There is a mixed use of tabs and spaces.

Comment (jrfnl) [inline WordPress/AbstractFunctionRestrictionsSniff.php]: Good catch, I hadn't noticed. It should be all spaces as it's in a comment, but the earlier code wasn't. Fixed now.

File: WordPress/AbstractFunctionRestrictionsSniff.php [hunk 2]
```diff
@@ -58,14 +58,19 @@
 	 * This method should be overridden in extending classes.
 	 *
 	 * Example: groups => array(
-	 * 	'lambda' => array(
-	 * 		'type'      => 'error' | 'warning',
-	 * 		'message'   => 'Use anonymous functions instead please!',
-	 * 		'functions' => array( 'eval', 'create_function' ),
-	 * 	)
+	 * 	   'lambda' => array(
+	 * 	       'type'      => 'error' | 'warning',
```
Comment (grappler) [inline WordPress/AbstractFunctionRestrictionsSniff.php]: There is still a tab here

Comment (jrfnl) [inline WordPress/AbstractFunctionRestrictionsSniff.php]: Sneaky little buggers...

**Pull request merged**